### PR TITLE
[SYCL] Add UUID and sub-, sub-sub- device info in `sycl-ls --verbose`

### PR DESCRIPTION
**Pull request by uditagarwal97**
Example `sycl-ls --verbose` output:

```
...
Platform [#1]:
    Version  : 1.3
    Name     : Intel(R) Level-Zero
    Vendor   : Intel(R) Corporation
    Devices  : 1
        Type              : gpu
        Version           : 12.60.7
        Name              : Intel(R) Data Center GPU Max 1550
        Vendor            : Intel(R) Corporation
        Driver            : 1.3.26918
        UUID              : 1341282141147000580000000
        Num SubDevices    : 2
        Num SubSubDevices : 8
        Aspects           : gpu fp16 fp64 ...
```

### CODE REVIEW

File: sycl/tools/sycl-ls/sycl-ls.cpp [hunk 1]
```diff
@@ -83,6 +116,23 @@ static void printDeviceInfo(const device &Device, bool Verbose,
     std::cout << Prepend << "Vendor     : " << DeviceVendor << std::endl;
     std::cout << Prepend << "Driver     : " << DeviceDriverVersion << std::endl;
 
+    // Get and print device UUID, if it is available.
+    if (Device.has(aspect::ext_intel_device_info_uuid)) {
+      auto UUID = Device.get_info<sycl::ext::intel::info::device::uuid>();
+      std::cout << Prepend << "UUID       : ";
+      for (int i = 0; i < 16; i++) {
```
Comment (RaviNarayanaswamy) [inline sycl/tools/sycl-ls/sycl-ls.cpp:125]: Should you be using ZE_MAX_DEVICE_UUID_SIZE instead of assuming size is 16

Comment (uditagarwal97) [reply sycl/tools/sycl-ls/sycl-ls.cpp:125]: That can be done. But, Isn't the UUID size fixed to 16: https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_device_info.md#device-uuid?

Comment (RaviNarayanaswamy) [reply sycl/tools/sycl-ls/sycl-ls.cpp:125]: If it is documented as 16 then ok.

**Review by RaviNarayanaswamy — APPROVED** [review]

File: sycl/test-e2e/Plugin/sycl-ls-uuid-subdevs.cpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+// As of now, ZEX_NUMBER_OF_CCS is not working with FLAT hierachy,
+// which is the new default on PVC.
+
+// RUN: env ONEAPI_DEVICE_SELECTOR="level_zero:*" env ZE_FLAT_DEVICE_HIERARCHY=COMPOSITE env ZEX_NUMBER_OF_CCS=0:4 sycl-ls --verbose >%t.default.out
+// RUN: FileCheck %s --input-file %t.default.out
+
+// CHECK: {{.*}}UUID              : {{.*}}
+// CHECK: {{.*}}Num SubDevices    : {{.*}}
+// CHECK-NEXT: {{.*}}Num SubSubDevices : {{.*}}
```
Comment (kbenzie) [inline sycl/test-e2e/Plugin/sycl-ls-uuid-subdevs.cpp:15]: nit: Missing new-line at the end of this file.

**Review by kbenzie — APPROVED** [review]

File: sycl/test-e2e/Plugin/sycl-ls-uuid-subdevs.cpp [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/* Test to check that sycl-ls is outputting UUID and number of sub and sub-sub
+ * devices. */
+// REQUIRES:  gpu, level_zero
```
Comment (cperkinsintel) [inline sycl/test-e2e/Plugin/sycl-ls-uuid-subdevs.cpp:3]: will this work on Gen12 ? Or is it PVC only?

**Review by cperkinsintel — APPROVED** [review]

File: sycl/test-e2e/Plugin/sycl-ls-uuid-subdevs.cpp (hunk 3 above)
Comment (uditagarwal97) [reply sycl/test-e2e/Plugin/sycl-ls-uuid-subdevs.cpp:3]: It should work on Gen12 as well. The test passes on Gen12 precommit: https://github.com/intel/llvm/actions/runs/9343215312/job/25714501971

**Comment by uditagarwal97** [comment]
Pre commit failure in Linux Gen12 is in `BFloat16/bfloat16_builtins.cpp` E2E test which is unrelated to this PR and reported in https://github.com/intel/llvm/issues/13790

**Comment by uditagarwal97** [comment]
@intel/llvm-gatekeepers The PR is ready!

File: sycl/tools/sycl-ls/sycl-ls.cpp (hunk 1 above)
Comment (mabraham) [reply sycl/tools/sycl-ls/sycl-ls.cpp:130]: Or just use a range-based for loop on UUID

**Pull request merged**